### PR TITLE
Bmoric/temporal cleaning cron

### PR DESCRIPTION
**Pull request by benmoriceau**
## What
Update the cron project in order to properly inject the temporal client in the scheduled service.

### CODE REVIEW

File: airbyte-cron/src/main/java/io/airbyte/cron/selfhealing/Temporal.java [hunk 1]
```diff
@@ -4,19 +4,27 @@
...
 import javax.inject.Singleton;
 import lombok.extern.slf4j.Slf4j;
 
 @Singleton
 @Slf4j
 public class Temporal {
 
+  @Inject
+  private TemporalClient temporalClient;
+
```
Comment (colesnodgrass) [inline airbyte-cron/src/main/java/io/airbyte/cron/selfhealing/Temporal.java:20]: I'm curious if we have stance on private field injections vs constructor/setter injection. My preference would be to convert this to use constructor injection.  I have found that private field injections tend to make code more difficult to test.

File: airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java [hunk 2]
```diff
@@ -257,6 +257,7 @@ private CancellationScope generateSyncWorkflowRunnable(final ConnectionUpdaterIn
         // silently ignore it.
         if (childWorkflowFailure.getCause() instanceof CanceledFailure) {
           // do nothing, cancellation handled by cancellationScope
+          log.info("");
```
Comment (colesnodgrass) [inline airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java:260]: Is there is missing log message here? If not, then this should be removed.

File: airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java [hunk 3]
```diff
@@ -257,6 +257,7 @@ private CancellationScope generateSyncWorkflowRunnable(final ConnectionUpdaterIn
...
         } else if (childWorkflowFailure.getCause()instanceof final ActivityFailure af) {
```
Comment (colesnodgrass) [inline airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java:261]: This is existing code, but there appears to be a formatting issue here, there should be a space before `instanceof`

File: airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java (hunk 2 above)
Comment (benmoriceau) [reply airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java:260]: Sorry I forget to remove it, It was to avoid having a PMD failure.

File: airbyte-cron/src/main/java/io/airbyte/cron/selfhealing/Temporal.java (hunk 1 above)
Comment (benmoriceau) [reply airbyte-cron/src/main/java/io/airbyte/cron/selfhealing/Temporal.java:20]: I don't have a strong preference but I agree that it will be easier to test.

File: airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java (hunk 2 above)
Comment (benmoriceau) [reply airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java]: I changed it to a suppress warning, I'll do it in another commit as well.

File: airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java (hunk 3 above)
Comment (benmoriceau) [reply airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java]: Fixed in https://github.com/airbytehq/airbyte/pull/16454

File: airbyte-cron/src/main/resources/application.yml [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+  security:
+    intercept-url-map:
+      - pattern: /**
+        httpMethod: GET
+        access:
+          - isAnonymous()
+  server:
+    port: 9000
+
+airbyte:
```
Comment (gosusnp) [inline airbyte-cron/src/main/resources/application.yml:13]: I am a bit confused about this part. It looks like we are starting to duplicate configs and a lot of the keys here feel irrelevant to the `airbyte-cron` itself.

File: airbyte-cron/src/main/resources/application-control.yml [hunk 5]
```diff
@@ -0,0 +1,40 @@
+datasources:
```
Comment (jdpgrailsdev) [inline airbyte-cron/src/main/resources/application-control.yml:1]: We shouldn't need this file in the cron service.

File: airbyte-cron/src/main/resources/application.yml (hunk 4 above)
Comment (jdpgrailsdev) [reply airbyte-cron/src/main/resources/application.yml:13]: @gosusnp Agreed.  We shouldn't need the full config from the `airbyte-workers` branch.  We should only need the config bits needed to configure the Temporal client.  I would start by commenting them all out and then restoring the ones that are actually used (Micronaut will complain at runtime that it can't find the property).

File: airbyte-cron/src/main/resources/application-control.yml (hunk 5 above)
Comment (benmoriceau) [reply airbyte-cron/src/main/resources/application-control.yml:1]: The issue here is that all the Beans are being initiated because the cron module depends on the worker one which has all the beans being declared.

File: airbyte-cron/src/main/resources/application.yml (hunk 4 above)
Comment (benmoriceau) [reply airbyte-cron/src/main/resources/application.yml:13]: @jdpgrailsdev @gosusnp I answered it here: https://github.com/airbytehq/airbyte/pull/16414#discussion_r966364804 I will bring it to the sit and work.

File: airbyte-cron/src/main/resources/application-control.yml (hunk 5 above)
Comment (benmoriceau) [reply airbyte-cron/src/main/resources/application-control.yml:1]: The eager initiation was set to true, I am removing it and will try to clean this list. Unfortunately we will need more config than what we thought.

Comment (jdpgrailsdev) [reply airbyte-cron/src/main/resources/application-control.yml:1]: Yeah...this is because we depend on `airbyte-workers`.  That causes Micronaut to find all singletons.  The solution, ultimately, is to extract the shared code to a library module that both can depend on.

Comment (benmoriceau) [reply airbyte-cron/src/main/resources/application-control.yml:1]: We sync up with @gosusnp and discover that we have ApplicationInitializer. I think that the best way to do it is to have the cron module to not depends on the worker module. I will try to extract the temporal client from it in another PR.

Comment (jdpgrailsdev) [reply airbyte-cron/src/main/resources/application-control.yml:1]: @benmoriceau Agreed...that is the best course here.  It will cause some conflict with the airbyte-workers Micronaut PR, but if you are able to extract and get that merged first, I can easily update the PR.

File: airbyte-webapp/src/components/base/Input/Input.tsx [hunk 6]
```diff
@@ -4,85 +4,16 @@ import classNames from "classnames";
...
-const getBackgroundColor = (props: IStyleProps) => {
-  if (props.error) {
-    return props.theme.greyColor10;
-  } else if (props.light) {
-    return props.theme.whiteColor;
-  }
-
-  return props.theme.greyColor0;
-};
+import styles from "./Input.module.scss";
```
Comment (lmossman) [inline airbyte-webapp/src/components/base/Input/Input.tsx:9]: @benmoriceau are these airbyte-webapp changes intended to be part of this PR? It seems like they are unrelated and maybe the result of a bad merge or something

Comment (benmoriceau) [reply airbyte-webapp/src/components/base/Input/Input.tsx]: @lmossman It is unrelated. My base branch is not master and every time the base is updated, unrelated changes appears :/. I merge it and the review will display the 6 changes that are actual changes.

**Comment by jdpgrailsdev** [comment]
@benmoriceau There appears to be a compile error in the branch:

```
import io.airbyte.workers.temporal.TemporalClient;
                                  ^
> Task :airbyte-cron:compileJava FAILED
/actions-runner/_work/airbyte/airbyte/airbyte-cron/src/main/java/io/airbyte/cron/selfhealing/Temporal.java:18: error: cannot find symbol
  private final TemporalClient temporalClient;
                ^
  symbol:   class TemporalClient
  location: class Temporal
/actions-runner/_work/airbyte/airbyte/airbyte-cron/src/main/java/io/airbyte/cron/selfhealing/Temporal.java:20: error: cannot find symbol
  public Temporal(@Named("temporalClient") final TemporalClient temporalClient) {
                                                 ^
  symbol:   class TemporalClient
  location: class Temporal
3 errors
```

**Comment by benmoriceau** [comment]
> @benmoriceau There appears to be a compile error in the branch:
> 
> ```
> import io.airbyte.workers.temporal.TemporalClient;
>                                   ^
> > Task :airbyte-cron:compileJava FAILED
> /actions-runner/_work/airbyte/airbyte/airbyte-cron/src/main/java/io/airbyte/cron/selfhealing/Temporal.java:18: error: cannot find symbol
>   private final TemporalClient temporalClient;
>                 ^
>   symbol:   class TemporalClient
>   location: class Temporal
> /actions-runner/_work/airbyte/airbyte/airbyte-cron/src/main/java/io/airbyte/cron/selfhealing/Temporal.java:20: error: cannot find symbol
>   public Temporal(@Named("temporalClient") final TemporalClient temporalClient) {
>                                                  ^
>   symbol:   class TemporalClient
>   location: class Temporal
> 3 errors
> ```

@jdpgrailsdev It was build on my local the build should now work

File: airbyte-cron/src/main/resources/application-control.yml (hunk 5 above)
Comment (benmoriceau) [reply airbyte-cron/src/main/resources/application-control.yml:1]: @jdpgrailsdev After merging and merging the extraction of the TemporalClient, I'll clean this file.

File: docker-compose.yaml [hunk 7]
```diff
@@ -183,14 +183,26 @@ services:
...
+      - JOB_ERROR_REPORTING_STRATEGY=${JOB_ERROR_REPORTING_STRATEGY}
+      - JOB_ERROR_REPORTING_SENTRY_DSN=${JOB_ERROR_REPORTING_SENTRY_DSN}
+      - LOCAL_DOCKER_MOUNT=${LOCAL_DOCKER_MOUNT}
+      - LOCAL_ROOT=${LOCAL_ROOT}
       - LOG_LEVEL=${LOG_LEVEL}
+      - INTERNAL_API_HOST=${INTERNAL_API_HOST}
       - POSTGRES_PWD=${DATABASE_PASSWORD}
       - POSTGRES_SEEDS=${DATABASE_HOST}
       - POSTGRES_USER=${DATABASE_USER}
-      - TEMPORAL_HISTORY_RETENTION_IN_DAYS=${TEMPORAL_HISTORY_RETENTION_IN_DAYS}
```
Comment (gosusnp) [inline docker-compose.yaml:192]: This is currently needed for the log cleaner.

File: airbyte-cron/build.gradle [hunk 8]
```diff
@@ -3,7 +3,27 @@ plugins {
...
+    implementation 'io.temporal:temporal-sdk:1.8.1'
+    implementation 'io.temporal:temporal-serviceclient:1.8.1'
+
+    implementation project(':airbyte-api')
+    implementation project(':airbyte-analytics')
     implementation project(':airbyte-config:config-models')
+    implementation project(':airbyte-config:config-persistence')
+    implementation project(':airbyte-db:db-lib')
+    implementation project(':airbyte-metrics:metrics-lib')
+    implementation (project(':airbyte-scheduler:scheduler-persistence')) {
```
Comment (jdpgrailsdev) [inline airbyte-cron/build.gradle:18]: @benmoriceau This can probably be replaced by the new module that @colesnodgrass made as part of the work to delete the scheduler module.

File: airbyte-cron/src/main/java/io/airbyte/cron/selfhealing/Temporal.java [hunk 9]
```diff
@@ -4,19 +4,27 @@
...
 
 @Singleton
 @Slf4j
 public class Temporal {
 
-  public Temporal() {
+  private final TemporalClient temporalClient;
+
+  public Temporal(@Named("temporalClient") final TemporalClient temporalClient) {
     log.info("Creating temporal self-healing");
```
Comment (jdpgrailsdev) [inline airbyte-cron/src/main/java/io/airbyte/cron/selfhealing/Temporal.java:21]: Can this info call be removed (or changed to debug)?

Comment (benmoriceau) [reply airbyte-cron/src/main/java/io/airbyte/cron/selfhealing/Temporal.java:21]: Changed to debug

File: airbyte-cron/build.gradle (hunk 8 above)
Comment (benmoriceau) [reply airbyte-cron/build.gradle:18]: Removed

**Review by gosusnp — APPROVED** [review]

**Review by jdpgrailsdev — APPROVED** [review]
:shipit:
